fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/RemoveAggregationFromAggTable.java

```diff
@@ -0,0 +1,120 @@
+        if (scanOperator.getProjection() != null) {
+            return false;
+        }
+        OlapTable olapTable = (OlapTable) scanOperator.getTable();
+
+        MaterializedIndexMeta materializedIndexMeta = olapTable.getIndexMetaByIndexId(scanOperator.getSelectedIndexId());
+        // must be Aggregation
+        if (!materializedIndexMeta.getKeysType().isAggregationFamily()) {
+            return false;
+        }
+        Set<String> keyColumnNames = Sets.newHashSet();
+        for (Column column : materializedIndexMeta.getSchema()) {
+            if (column.isKey()) {
+                keyColumnNames.add(column.getName().toLowerCase());
+            }
+        }
+
+        // group by keys contain partition columns and distribution columns
+        PartitionInfo partitionInfo = olapTable.getPartitionInfo();
+        List<String> partitionColumnNames = Lists.newArrayList();
+        if (partitionInfo instanceof RangePartitionInfo) {
+            RangePartitionInfo rangePartitionInfo = (RangePartitionInfo) partitionInfo;
+            partitionColumnNames.addAll(rangePartitionInfo.getPartitionColumns().stream()
+                    .map(column -> column.getName().toLowerCase()).collect(Collectors.toList()));
+        }
+
+        List<String> distributionColumnNames = olapTable.getDistributionColumnNames().stream()
+                .map(String::toLowerCase).collect(Collectors.toList());
+
+        LogicalAggregationOperator aggregationOperator = (LogicalAggregationOperator) input.getOp();
+        for (Map.Entry<ColumnRefOperator, CallOperator> entry : aggregationOperator.getAggregations().entrySet()) {
+            if (unsupportedFunctionNames.contains(entry.getValue().getFnName().toLowerCase())) {
+                return false;
+            }
+        }
+        List<String> groupKeyColumns = aggregationOperator.getGroupingKeys().stream()
+                .map(columnRefOperator -> columnRefOperator.getName().toLowerCase()).collect(Collectors.toList());
+        return groupKeyColumns.containsAll(keyColumnNames)
+                && groupKeyColumns.containsAll(partitionColumnNames)
+                && groupKeyColumns.containsAll(distributionColumnNames);
+    }
+
+    @Override
+    public List<OptExpression> transform(OptExpression input, OptimizerContext context) {
+        LogicalAggregationOperator aggregationOperator = (LogicalAggregationOperator) input.getOp();
+        Map<ColumnRefOperator, ScalarOperator> projectMap = Maps.newHashMap();
+        aggregationOperator.getGroupingKeys().forEach(g -> projectMap.put(g, g));
+        for (Map.Entry<ColumnRefOperator, CallOperator> entry : aggregationOperator.getAggregations().entrySet()) {
+            // in this case, CallOperator must have only one child ColumnRefOperator
+            projectMap.put(entry.getKey(), entry.getValue().getChild(0));
+        }
+        if (aggregationOperator.getProjection() != null) {
```

Comment:
call this rule after MaterializedViewRewrite, and you don't need handle projection

Reply:
I test this way, it is not ok.